proto/api/v1/auth_service.proto

```diff
@@ -20,21 +21,149 @@ message UserAttribute {
+        // This supports regular expressions of RE2 format, see more here:
+        // https://github.com/google/re2/wiki/Syntax.
+        string value          = 2;
+        // The role which should be issued when the key and value match for a particular identity token.
+        string role           = 3;
+    }
+    repeated Mapping mappings = 4;
+
+    // The issuer configuration for a generic OIDC issuer of the identity token.
+    // For the GitHub Action type, no configuration is required and the generic issuer will be ignored.
+    message GenericIssuer {
+        string issuer                       = 1;
+    }
+    oneof IssuerConfig {
+        GenericIssuer generic_issuer_config = 5;
+    }
+}
+
+message ListAuthMachineToMachineConfigResponse {
+    repeated AuthMachineToMachineConfig configs = 1;
+}
+
+message GetAuthMachineToMachineConfigResponse {
+    AuthMachineToMachineConfig config = 1;
+}
+
+message AddAuthMachineToMachineConfigRequest {
+    AuthMachineToMachineConfig config = 1;
+}
+
+message AddAuthMachineToMachineConfigResponse {
+    AuthMachineToMachineConfig config = 1;
+}
+
+message UpdateAuthMachineToMachineConfigRequest {
+    AuthMachineToMachineConfig config = 1;
+}
+
+message UpdateAuthMachineToMachineConfigResponse {
+    AuthMachineToMachineConfig config = 1;
+}
+
+message ExchangeAuthMachineToMachineTokenRequest {
+    // Identity token that is supposed to be exchanged.
+    string id_token = 1;
+}
 
-    storage.AuthProvider   auth_provider   = 5;
-    storage.UserInfo       user_info       = 6;
-    repeated UserAttribute user_attributes = 7;
+message ExchangeAuthMachineToMachineTokenResponse {
+    // The exchanged access token.
+    string access_token = 1;
 }
 
 service AuthService {
+    // GetAuthStatus returns the status for the current client.
     rpc GetAuthStatus(Empty) returns (AuthStatus) {
         option (google.api.http) = {
             get: "/v1/auth/status"
         };
     }
+
+    // ListAuthMachineToMachineConfigs lists the available auth machine to machine configs.
+    rpc ListAuthMachineToMachineConfigs(Empty) returns (ListAuthMachineToMachineConfigResponse) {
+        option (google.api.http) = {
+            get: "/v1/auth/m2m"
+        };
+    }
+
+    // GetAuthMachineToMachineConfig retrieves the specific auth machine to machine config.
+    rpc GetAuthMachineToMachineConfig(ResourceByID) returns (GetAuthMachineToMachineConfigResponse) {
+        option (google.api.http) = {
+            get: "/v1/auth/m2m/{id}"
+        };
+    }
+
+    // AddAuthMachineToMachineConfig creates a new auth machine to machine config.
+    rpc AddAuthMachineToMachineConfig(AddAuthMachineToMachineConfigRequest) returns (AddAuthMachineToMachineConfigResponse) {
+        option (google.api.http) = {
+            post: "/v1/auth/m2m"
+            body: "*"
+        };
+    }
+
+    // UpdateAuthMachineToMachineConfig updates an existing auth machine to machine config.
```

Comment:
Following up on our [previous thread](https://github.com/stackrox/stackrox/pull/8145#discussion_r1363675148), this must say that a config will be created if one does not exist.